app/components/exp-lookit-iframe/doc.rst

```diff
@@ -0,0 +1,68 @@
+.. _exp-lookit-iframe:
+
+exp-lookit-iframe
+==============================================
+
+Overview
+------------------
+
+A frame to display external websites, such as surveys, to the participant. 
```

Comment:
Here's my suggestion for additional text in the overview section, but @mekline may have other thoughts/suggestions!

> A frame to display external websites within a Lookit study, such as surveys or tasks hosted on other platforms. A 'Next' button is displayed beneath the iframe box that allows the participant to continue to the next Lookit frame.
> 
> This frame is useful for embedding an externally-hosted webpage in the _middle_ of a Lookit study. Importantly, the Lookit study has no way of 'knowing' what sorts of interactions the participant has, or has not had, on the external website. For example, there is no way to ensure that the participant has made a response on the external site before they click the 'Next' button to continue on with the Lookit study. For this reason, if your study methods allow it, we suggest moving any externally-hosted surveys/tasks to the _end_ of the Lookit study. You can automatically direct participants to another website at the end of the Lookit study using the study's 'Exit URL' (link to documentation: https://lookit.readthedocs.io/en/develop/researchers-set-study-fields.html#exit-url).
> 
> If you do need to embed an external website in the middle of a Lookit study, this frame includes the `optionalText` parameter, which allows you to add text underneath the iframe box. You can use this text to help ensure the participant completes everything in the iframe box before clicking the frame's 'Next' button, e.g. "Please make sure that you see the message 'Your response has been submitted' in the box above before clicking the next button below."

Reply:
Couple of additions to address an important use case, getting video data on participants completing a task on a website not providing video recording! I'm leaving some details out because I'm not sure exactly how single-frame recording does/doesn't happen on this frame (vs. using the start-session-recording frame), can someone fill in please? 

****
A frame to display external websites within a Lookit study, such as surveys or tasks hosted on other platforms. A 'Next' button is displayed beneath the iframe box that allows the participant to continue to the next Lookit frame.

This frame is useful for embedding an externally-hosted webpage in the middle of a Lookit study. You might do this when you have a preexisting task you have already been using for data collection that you would like to incorporate into an internal study, or when you have a task that is outside of the capabilities of the Lookit frameplayer. This is especially useful when the external task happens in the middle of a session (e.g. Lookit-based video consent, external task, internal task, Lookit-based exit survey.) 

If your study methods allow it, an alternate suggestion is to move any externally-hosted surveys/tasks to the end of the Lookit study. In place of this frame, you can automatically direct participants to another website at the end of the Lookit study using the study's 'Exit URL' (link to documentation: https://lookit.readthedocs.io/en/develop/researchers-set-study-fields.html#exit-url).

Another reason to use this frame is if you want to capture webcam video of participants while they are completing a task on a site that doesn't provide webcam monitoring. While the participant is interacting with the externally-hosted webpage, you can record webcam video for this frame by **[DETAILS]**. Note that any timing information collected by the external site (e.g. externally recorded response time after the external page loads) will not necessarily correspond precisely to the Lookit frame timing information or exact onset of the resulting video file (link to documentation: https://lookit.readthedocs.io/en/develop/researchers-lag-issues.html). 

More generally, remember that the Lookit study has no way of 'knowing' what sorts of interactions the participant has, or has not had, on the external website. For example, there is no way to ensure that the participant has made a response on the external site before they click the 'Next' button to continue on with the Lookit study. 

This frame includes the optionalText parameter, which allows you to add text underneath the iframe box. You can use this text to help ensure the participant completes everything in the iframe box before clicking the frame's 'Next' button, e.g. "Please make sure that you see the message 'Your response has been submitted' in the box above before clicking the next button below."

Reply:
@okaycj Flagging some updates to Becky's excellent text here, I am sure there's a better way to do this on GH than a random comment but I've only just remembered that!

Reply:
LOL!  Thanks.  I'll make the update.